server/channels/api4/config.go

```diff
@@ -406,12 +401,8 @@ func makeFilterConfigByPermission(accessType filterType) func(c *Context, struct
 
 		tagPermissions := strings.Split(structField.Tag.Get("access"), ",")
 
-		// If there are no access tag values and the role has manage_system, no need to continue
-		// checking permissions.
-		if len(tagPermissions) == 0 {
-			if c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
-				return true
-			}
+		if c.AppContext.Session().IsUnrestricted() {
```

Comment:
Removing the redundant code, and then /adding/ the unrestricted check before since we examine the `ExperimentalSettings.RestrictSystemAdmin` inline below. (This is case where we can't use the newly added permissions helper.)